zebra/rt_socket.c

```diff
@@ -312,28 +312,32 @@ enum zebra_dplane_result kernel_route_update(struct zebra_dplane_ctx *ctx)
 	type = dplane_ctx_get_type(ctx);
 	old_type = dplane_ctx_get_old_type(ctx);
 
-	frr_with_privs(&zserv_privs) {
+	flag = dplane_ctx_get_flag(ctx);
+	old_flag = dplane_ctx_get_old_flag(ctx);
+	if (IS_ZEBRA_DEBUG_KERNEL)
+		zlog_debug("%s: rt_socket, zebra ctx flag = %d, old flag = %d",flag , old_flag);
 
+	frr_with_privs(&zserv_privs) {
 		if (dplane_ctx_get_op(ctx) == DPLANE_OP_ROUTE_DELETE) {
-			if (!RSYSTEM_ROUTE(type))
+			if (!RSYSTEM_ROUTE(type) && !CHECK_FLAG(flag, DPLANE_RINFO_FLAG_NO_KERNEL))
```

Comment:
again why not just do this first? instead of having to have a check flag call in every possible combination